salt/modules/virt.py

```diff
@@ -1179,23 +1214,73 @@ def _gen_vol_xml(
     return template.render(**context)
 
 
-def _gen_net_xml(name, bridge, forward, vport, tag=None, ip_configs=None):
+def _gen_net_xml(
+    name,
+    bridge,
+    forward,
+    vport,
+    tag=None,
+    ip_configs=None,
+    mtu=None,
+    domain=None,
+    nat=None,
+    interfaces=None,
+    addresses=None,
+    physical_function=None,
+    dns=None,
+):
     """
     Generate the XML string to define a libvirt network
     """
+    vport_context = vport
+    if isinstance(vport, str):
+        vport_context = {"type": vport}
+
+    tag_context = tag
+    if isinstance(tag, str) or isinstance(tag, int):
```

Comment:
**suggestion** isinstance can take a tuple:

```
if isinstance(tag, (str,int)):
    tag = {"tags": [...]}
```

Reply:
thanks for the tip, I wasn't aware of that!